File: datamodel/src/main/java/datamodel/clinical/PatientDetails.java
package datamodel.clinical;

import java.time.LocalDate;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PatientDetails {

    @NotNull
    public abstract Gender gender();

    public abstract int birthYear();

    @NotNull
    public abstract LocalDate registrationDate();

    @Nullable
    public abstract LocalDate questionnaireDate();

    @Nullable
    public abstract String otherMolecularPatientId();

}

File: algo/src/main/kotlin/com/hartwig/actin/clinical/datamodel/PatientDetails.kt
package com.hartwig.actin.clinical.datamodel

import java.time.LocalDate

data class PatientDetails(
    val gender: Gender,
    val birthYear: Int,
    val registrationDate: LocalDate,
    val questionnaireDate: LocalDate? = null,
    val otherMolecularPatientId: String? = null
)
